### DISCUSSION TITLE
Navigating through Jupyter Notebook source code

### DISCUSSION THREAD

**Discussion by l3atjin** [jupyter/notebook#6653] [Q&A]
Hi all,

For my background, I am a computer science student who is looking to make a contribution to Jupyter Notebook.

I've picked up an issue (https://github.com/jupyter/notebook/issues/6556) but I'm having a hard time navigating through the source. I was able to build and run notebook from my local machine but I'm confused about where to start. 

I was reading various documentation on notebook and am generally confused about how all of these extensions work. I'm looking to find the front end render files and understand them as a starter. Any help or direction would be greatly appreciated!

**Comment by jtpio** [comment]
Hi @l3atjin,

> I've picked up an issue (#6556) but I'm having a hard time navigating through the source. I was able to build and run notebook from my local machine but I'm confused about where to start.

The current `main` branch of the `notebook` repo contains the code for the next major version of Jupyter Notebook (Notebook 7). It is based on JupyterLab components an extensions which can explain why some parts might not be visible since they are directly reused from JupyterLab.

The main application is built in the `app` folder: https://github.com/jupyter/notebook/tree/main/app. Its package.json contains the list of extensions included in the build: https://github.com/jupyter/notebook/blob/0b99dcc820bcfb01455b92bf8d86cfc3cb3b88fc/app/package.json#L181

The Notebook application also defines a couple more extensions to handle some differences with stock JupyterLab. They are located in the `packages` folder and also included in the build.

> I was reading various documentation on notebook and am generally confused about how all of these extensions work. I'm looking to find the front end render files and understand them as a starter. Any help or direction would be greatly appreciated!

Normally everything should be documented in the contributing guide: https://github.com/jupyter/notebook/blob/main/CONTRIBUTING.md. Are you stuck on a particular step, or is mostly a question about code organization?

As an alternative you should also be able to an cloud development environment like Gitpod to avoid having to install anything locally: https://gitpod.io/
